src/collectives/device/gen_rules.sh

```diff
@@ -21,10 +24,17 @@ for base in sendrecv all_reduce all_gather broadcast reduce reduce_scatter; do
     dtn=0
     # Order must match that of the ncclDataType_t enum
     for dt in ${datatypes}; do
-      echo "${dir}/${base}_${op}_${dt}.o : ${base}.cu ${dir}/${base}.dep"
+      # Generate a unique filename for each compilation unit,
+      # otherwise the __nv_module_id may conflict at link time
+      echo "${dir}/${base}_${opn}_${dtn}.cu :"
```

Comment:
I get the point of using #include here, but all those .cu files are almost empty, all the code is in the .h file. Therefore, it's much simpler to just copy them: that avoids the CURDIR definition above.